tasks/add_odl_deb_repo.yml

```diff
@@ -0,0 +1,18 @@
+- name: Add the repository key to apt
+  apt_key:
+    url={{ deb_key }}
+    state=present
+
+- name: Add ODL deb repo
+  apt_repository:
+    repo={{ deb_repo_url }}
+    state=present
+    filename='opendaylight'
```

Comment:
The source filenames end in .list. So, the repo is added in /etc/apt/sources.list.d/opendaylight.list.
    'filename' param docs: Sets the name of the source list file in sources.list.d. Defaults to a file name based on the repository source url. The .list extension will be automatically added.
See: http://docs.ansible.com/ansible/apt_repository_module.html